src/main/scala/chisel3/util/experimental/decode/decoder.scala

```diff
@@ -30,6 +31,7 @@ object decoder extends LazyLogging {
       val (plaInput, plaOutput) =
         pla(minimizedTable.table.toSeq, BitPat(minimizedTable.default.value.U(minimizedTable.default.getWidth.W)))
 
+      if (!plaOutput.isSynthesizable) Builder.error(s"Using DecodeTableAnnotation on non-hardware value $plaOutput")
```

Comment:
Since this is checking the output of one of our own APIs rather than user input, I don't think the standard `Builder.error` is the right check here (that's more for user-error). I think an `assert` would be better just as a sanity check.